app/views/LocationController.js

```diff
@@ -13,102 +13,90 @@ import { SvgXml } from 'react-native-svg';
   }
 
-  componentWillUnmount() {
-    console.log('unmounting component');
-    AppState.removeEventListener('change', this.handleAppStateChange);
-    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
-    this.unsubscribe();
-  }
-
-  handleBackPress = () => {
-    BackHandler.exitApp(); // works best when the goBack is async
-    return true;
-  };
+  const [location, setLocation] = useState({
+    canTrack: true,
+    reason: '',
+    hasPotentialExposure: false,
+  });
 
-  getSettings() {
+  const SettingsIcon = () => {
     return (
       <TouchableOpacity
         style={styles.settingsContainer}
         onPress={() => {
-          this.props.navigation.navigate('SettingsScreen');
+          props.navigation.navigate('SettingsScreen');
         }}>
         {/* Is there is a reason there's this imageless image tag here? Can we delete it? */}
-        <Image resizeMode={'contain'} />
+        <Image />
         <SvgXml xml={settingsIcon} width={30} height={30} color='white' />
       </TouchableOpacity>
     );
-  }
+  };
 
-  async updateStateInfo() {
+  const updateStateInfo = async () => {
     const state = await LocationServices.checkStatus();
-    this.setState({ location: state });
-  }
+    setLocation(state);
+  };
+
+  const handleBackPress = () => {
+    BackHandler.exitApp(); // works best when the goBack is async
+    return true;
+  };
+
+  useEffect(() => {
+    updateStateInfo();
+    // refresh state if user backgrounds app
+    AppState.addEventListener('change', () => {
+      updateStateInfo();
+    });
+    // refresh state if settings change
+    const unsubscribe = props.navigation.addListener('focus', () => {
+      updateStateInfo();
+    });
+    // handle back press
+    BackHandler.addEventListener('hardwareBackPress', handleBackPress);
 
-  render() {
-    const canTrack = this.state.location.canTrack;
-    const reason = this.state.location.reason;
-    const crossedPaths = this.state.location.crossedPaths;
-    let page;
+    return () => {
+      AppState.removeEventListener('change', updateStateInfo());
+      BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
+      unsubscribe();
+    };
+  }, []);
 
-    if (canTrack) {
-      if (crossedPaths) {
-        page = <ExposurePage />;
-      } else {
-        page = <DefaultPage />;
-      }
+  let page;
+
+  if (location.canTrack) {
+    if (location.hasPotentialExposure) {
+      page = <ExposurePage />;
     } else {
-      if (reason === 'USER_OFF') {
-        page = <OffPage />;
-      } else {
-        page = <UnknownPage />;
-      }
+      page = <DefaultPage />;
+    }
+  } else {
+    if (location.reason === Reason.USER_OFF) {
+      page = <OffPage />;
+    } else {
+      page = <UnknownPage />;
     }
-
-    return (
-      <View style={styles.backgroundImage}>
-        {page}
-        {this.getSettings()}
-      </View>
-    );
   }
-}
+
+  return (
+    <View style={styles.backgroundImage}>
```

Comment:
Here is a good place to switch themes. `<Theme>` handles the text and icon button color for child components.

`<Theme use={getTheme}>`

And `getTheme` can switch between `violet` for normal modes and `charcoal` when there is an exposure etc